compressor/compressor.cpp

```diff
@@ -0,0 +1,165 @@
+#include <fstream>
+#include <deque>
+#include "compressor.h"
+#include "BitCashier.h"
+#include "OffsetAndLength.h"
+#include <iostream>//
+
+
+int slide_window(unsigned int index, std::deque<Byte>& window, std::ifstream& in, const std::streampos& end_of_file)
+{
+  while (index > window.size() / 2 && window.size() > window_width / 2)
+ {
+    if (in.tellg() < end_of_file)
+    {
+      window.push_back(in.get());
+      window.pop_front();
+      --index;
+    }
+    else
+    {
+      window.pop_front();
+      --index;
+    }
+  }
+
+  return index;
+}
+
+bool find_a_match(const std::deque<Byte>& window, const unsigned int i, OffsetAndLength& off_len)
+{
+  int start = 0;
+  bool checking = false;
+
+  for (unsigned int k = 0; k < i; ++k)
+  {
+    if (window[k] == window[i] && checking == false)
+    {
+      checking = true;
+      start = k;
+    }
+    else if (checking == true && (k - start == max_match_length ||window.size() == i + k - start || window[k] != window[i + k - start]))
+    {
+      checking = false;
+      if ((unsigned int)(off_len.get_length()) < k - start)
+        off_len.set(i - start, k - start);
+
+      if (off_len.get_length() == max_match_length)
+        break;      
+
+      k = start + 1;
+      start = 0;
+    }
+  }
+
+  if (off_len.get_offset() == 0) put_error_and_exit("Offset is impossible\n");//
```

Comment:
If this is impossible it should be an assert. More over such a long if statement better to make two line. But this particular one should be an assert.